### ISSUE TITLE
Nehalem: Programmable Ratio Limit for Turbo Mode

### ISSUE THREAD

**Issue by cyring** [cyring/CoreFreq#62]
With the Intel i7-920, bit 28 of MSR_PLATFORM_INFO [0xce] says the opposite of the queried value.
See the model-specific register specification in the [SDM](https://github.com/cyring/CoreFreq/wiki/Documentation) : Table 2-14. MSRs in Processors Based on Intel® Microarchitecture Code Name Nehalem

**Comment by cyring** [comment]
Bits decoded of the Intel i7-920 Processor.
Issue appears in "Ratio Limited".
```
Processor                               [Intel(R) Core(TM) i7 CPU 920 @ 2.67GHz]
|- Architecture                                             [Nehalem/Bloomfield]
|- Vendor ID                                                      [GenuineIntel]
|- Signature                                                            [ 06_1A]
|- Stepping                                                             [     4]
|- Microcode                                                            [    18]
|- Online CPU                                                           [  8/8 ]
|- Base Clock                                                           [152.50]
|- Ratio Limited                                                        [UNLOCK]
|- Frequency            (Mhz)                      Ratio                        
                 Min   1829.94                    [  12 ]                       
                 Max   3049.91                    [  20 ]                       
|- Factory                                                                      
                       2670                       [  20 ]                       
|- Turbo Boost                                                                  
                  1C   3354.90                    <  22 >                       
                  2C   3202.40                    <  21 >                       
                  3C   3202.40                    <  21 >                       
                  4C   3202.40                    <  21 >                       
|- Uncore                                                                       
                 Min   1829.94                    [  12 ]                       
                 Max   2439.93                    [  16 ]                       
                                                                                
ISA Extensions:                                                                 
|- 3DNow!/Ext [N,N]        AES [N]   AVX/AVX2 [N/N]   BMI1/BMI2 [N/N] 
|- CLFSH        [Y]       CMOV [Y]      CMPXCH8 [Y]      CMPXCH16 [Y] 
|- F16C         [N]        FPU [Y]         FXSR [Y]     LAHF/SAHF [Y] 
|- MMX/Ext    [Y/N]    MONITOR [Y]        MOVBE [N]      PCLMULDQ [N] 
|- POPCNT       [Y]     RDRAND [N]       RDTSCP [Y]           SEP [Y] 
|- SGX          [N]        SSE [Y]         SSE2 [Y]          SSE3 [Y] 
|- SSSE3        [Y] SSE4.1/4A [Y/N]      SSE4.2 [Y]       SYSCALL [Y] 
                                                                                
Features:                                                                       
|- 1 GB Pages Support                                      1GB-PAGES   [Missing]
|- 100 MHz multiplier Control                            100MHzSteps   [Missing]
|- Advanced Configuration & Power Interface                     ACPI   [Present]
|- Advanced Programmable Interrupt Controller                   APIC   [Present]
|- Core Multi-Processing                                  CMP Legacy   [Missing]
|- L1 Data Cache Context ID                                  CNXT-ID   [Missing]
|- Direct Cache Access                                           DCA   [Missing]
|- Debugging Extension                                            DE   [Present]
|- Debug Store & Precise Event Based Sampling               DS, PEBS   [Present]
|- CPL Qualified Debug Store                                  DS-CPL   [Present]
|- 64-Bit Debug Store                                         DTES64   [Present]
|- Fast-String Operation                                Fast-Strings   [Missing]
|- Fused Multiply Add                                       FMA|FMA4   [Missing]
|- Hardware Lock Elision                                         HLE   [Missing]
|- Long Mode 64 bits                                         IA64|LM   [Present]
|- LightWeight Profiling                                         LWP   [Missing]
|- Machine-Check Architecture                                    MCA   [Present]
|- Model Specific Registers                                      MSR   [Present]
|- Memory Type Range Registers                                  MTRR   [Present]
|- OS-Enabled Ext. State Management                          OSXSAVE   [Missing]
|- Physical Address Extension                                    PAE   [Present]
|- Page Attribute Table                                          PAT   [Present]
|- Pending Break Enable                                          PBE   [Present]
|- Process Context Identifiers                                  PCID   [Missing]
|- Perfmon and Debug Capability                                 PDCM   [Present]
|- Page Global Enable                                            PGE   [Present]
|- Page Size Extension                                           PSE   [Present]
|- 36-bit Page Size Extension                                  PSE36   [Present]
|- Processor Serial Number                                       PSN   [Missing]
|- Restricted Transactional Memory                               RTM   [Missing]
|- Safer Mode Extensions                                         SMX   [Missing]
|- Self-Snoop                                                     SS   [Present]
|- Time Stamp Counter                                            TSC [Invariant]
|- Time Stamp Counter Deadline                          TSC-DEADLINE   [Missing]
|- Virtual Mode Extension                                        VME   [Present]
|- Virtual Machine Extensions                                    VMX   [Present]
|- Extended xAPIC Support                                     x2APIC   [Missing]
|- Execution Disable Bit Support                              XD-Bit   [Present]
|- XSAVE/XSTOR States                                          XSAVE   [Missing]
|- xTPR Update Control                                          xTPR   [Present]
                                                                                
Technologies:                                                                   
|- System Management Mode                                   SMM-Dual       [ ON]
|- Hyper-Threading                                               HTT       [ ON]
|- SpeedStep                                                    EIST       < ON>
|- Dynamic Acceleration                                          IDA       [ ON]
|- Turbo Boost                                                 TURBO       < ON>
|- Virtualization                                                VMX       [ ON]
   |- I/O MMU                                                   VT-d       [ ON]
   |- Hypervisor                                                           [OFF]
                                                                                
Performance Monitoring:                                                         
|- Version                                                        PM       [  3]
|- Counters:          General                   Fixed                           
|                     4 x 48 bits             3 x 48 bits                       
|- Enhanced Halt State                                           C1E       <OFF>
|- C1 Auto Demotion                                              C1A       <OFF>
|- C3 Auto Demotion                                              C3A       <OFF>
|- C1 UnDemotion                                                 C1U       <OFF>
|- C3 UnDemotion                                                 C3U       <OFF>
|- Frequency ID control                                          FID       [OFF]
|- Voltage ID control                                            VID       [OFF]
|- P-State Hardware Coordination Feedback                MPERF/APERF       [ ON]
|- Hardware-Controlled Performance States                        HWP       [OFF]
|- Hardware Duty Cycling                                         HDC       [OFF]
|- Package C-State                                                              
   |- Configuration Control                                   CONFIG   [ UNLOCK]
   |- Lowest C-State                                           LIMIT   <      0>
   |- I/O MWAIT Redirection                                  IOMWAIT   < ENABLE>
   |- Max C-State Inclusion                                    RANGE   <      7>
|- MWAIT States:      C0      C1      C2      C3      C4                        
|                      0       2       1       1       0                        
|- Core Cycles                                                         [Present]
|- Instructions Retired                                                [Present]
|- Reference Cycles                                                    [Present]
|- Last Level Cache References                                         [Present]
|- Last Level Cache Misses                                             [Present]
|- Branch Instructions Retired                                         [Present]
|- Branch Mispredicts Retired                                          [Missing]
                                                                                
Power & Thermal Monitoring:                                                     
|- Clock Modulation                                             ODCM   <Disable>
   |- DutyCycle                                                        <  0.00%>
|- Power Management                                         PWR MGMT   [   LOCK]
   |- Energy Policy                                        Bias Hint   [      0]
|- Junction Temperature                                        TjMax   [    100]
|- Digital Thermal Sensor                                        DTS   [Present]
|- Power Limit Notification                                      PLN   [Missing]
|- Package Thermal Management                                    PTM   [Missing]
|- Thermal Monitor 1                                         TM1|TTP   [ Enable]
|- Thermal Monitor 2                                         TM2|HTC   [Present]
|- Units                                                                        
   |- Power                                               watt   [  0.000000000]
   |- Energy                                             joule   [  0.000000000]
   |- Window                                            second   [  0.000000000]
```

**Comment by cyring** [comment]
The above commit allows to override the Processor overclocking capability when registers are miss specified.

**Comment by cyring** [comment]
Issue closed in favor of issue #67
